apps/api/src/app/events/usecases/process-subscriber/process-subscriber.usecase.ts

```diff
@@ -1,161 +1,56 @@
+    const subscriberEntity = await this.getSubscriber(environmentId, organizationId, subscriber);
 
-      const providerId: string | undefined = await this.getProviderId(command, step.template.type);
-      jobs.push({
-        identifier: command.identifier,
-        payload: command.payload,
-        overrides: command.overrides,
-        step,
-        transactionId: command.transactionId,
-        _notificationId: notification._id,
-        _environmentId: command.environmentId,
-        _organizationId: command.organizationId,
-        _userId: command.userId,
-        _subscriberId: subscriber._id,
-        status: JobStatusEnum.PENDING,
-        _templateId: notification._templateId,
-        digest: step.metadata,
-        type: step.template.type,
-        providerId: providerId,
-        ...(actorSubscriber && { _actorId: actorSubscriber._id }),
-      });
+    if (subscriberEntity === null) {
+      return undefined;
     }
 
-    return jobs;
-  }
-
-  private async getProviderId(command: ProcessSubscriberCommand, stepType: StepTypeEnum): Promise<string | undefined> {
-    const channelType = STEP_TYPE_TO_CHANNEL_TYPE.get(stepType);
-    if (!channelType) return;
-    const integrations = await this.getDecryptedIntegrations.execute(
-      GetDecryptedIntegrationsCommand.create({
-        channelType: channelType,
-        active: true,
-        organizationId: command.organizationId,
-        environmentId: command.environmentId,
-        userId: command.userId,
-      })
-    );
-    const integration = integrations[0];
-
-    return integration?.providerId ?? InAppProviderIdEnum.Novu;
-  }
-  @Cached(CacheKeyPrefixEnum.NOTIFICATION_TEMPLATE)
-  private async getNotificationTemplate({ _id, environmentId }: { _id: string; environmentId: string }) {
-    return await this.notificationTemplateRepository.findById(_id, environmentId);
+    return subscriberEntity;
   }
 
   private async getSubscriber(
-    command: Pick<ProcessSubscriberCommand, 'environmentId' | 'organizationId'>,
-    subscriberPayload
+    environmentId: string,
+    organizationId: string,
+    subscriberPayload: ISubscribersDefine
   ): Promise<SubscriberEntity> {
     const subscriber = await this.subscriberRepository.findBySubscriberId(
-      command.environmentId,
+      environmentId,
       subscriberPayload.subscriberId
     );
 
     if (subscriber && !subscriberNeedUpdate(subscriber, subscriberPayload)) {
       return subscriber;
     }
 
-    return await this.createOrUpdateSubscriber(command, subscriberPayload, subscriber);
+    return await this.createOrUpdateSubscriber(environmentId, organizationId, subscriberPayload, subscriber);
   }
 
   private async createOrUpdateSubscriber(
-    command: Pick<ProcessSubscriberCommand, 'environmentId' | 'organizationId'>,
+    environmentId: string,
+    organizationId: string,
     subscriberPayload,
-    subscriber: SubscriberEntity | null
-  ) {
+    // TODO: Getting rid of this null would be amazing
```

Comment:
after we do a `findBySubscriberId` we should have a check and throw if null

Reply:
If we throw we would break the loop. That would mean that some notifications will be processed from the trigger but not all, leading to inconsistency. Unless we set up a rollback for the full trigger, for now I think it is better to leave the current behaviour of sending to the existing subscribers the notifications, and just skip it for non existent subscribers or failed subscribers.
Something worth to bring to tech summit to see what's the best direction we want to have for best fault tolerance. 👍🏻

Reply:
ahhh ok sorry I was confused about this TODO comment, I think it's not possible to get rid of that null type in this function then because the subscriber is created in the `createSubscriberUsecase.execute` call when it's null...